### PR TITLE
feat: Add comprehensive SBOM generation for supply chain transparency'

### PR DESCRIPTION
**Pull request by Aharshi3614**
## Description
This PR adds automated Software Bill of Materials (SBOM) generation to TENET-AI for supply chain transparency and security compliance.

### 📊 Dependencies Captured
| Type | Count | Ecosystem |
|------|-------|-----------|
| Python Packages | 18 | PyPI (FastAPI, scikit-learn, pandas, etc.) |
| Node.js Packages | 17 | npm (React, Vite, Recharts, etc.) |
| **Total** | **35** | - |

### ✅ What's Implemented
- Generate SBOM in **CycloneDX v1.5** format (OWASP standard)
- Generate SBOM in **SPDX v2.3** format (Linux Foundation standard)
- Support both **Python** (pip) and **Node.js** (npm) dependencies
- **GitHub Actions workflow** for automatic generation
- **Attach SBOMs to releases** as artifacts
- **Documentation** in `docs/security/sbom.md`

### 📁 Files Added
- `scripts/generate_sbom.py` - Main SBOM generation script
- `.github/workflows/sbom.yml` - CI/CD automation
- `docs/security/sbom.md` - Security documentation

### 🚀 How It Works
1. **Manual:** `python scripts/generate_sbom.py`
2. **Automatic:** Triggers on release creation, tag push (v*), or manual dispatch

### 🔍 Example Output
```json
{
  "bomFormat": "CycloneDX",
  "specVersion": "1.5",
  "components": [
    {"name": "fastapi", "version": "0.115.0"},
    {"name": "react", "version": "19.2.0"}
  ]
}
```

### 📋 Task Checklist

- [x] Generate SBOM in CycloneDX format (v1.5)
- [x] Generate SBOM in SPDX format (v2.3)
- [x] Support Python dependencies (18 packages found)
- [x] Support Node.js dependencies (17 packages found)
- [x] Integrate SBOM generation into CI pipeline
- [x] Publish SBOM as release artifact
- [x] Document supply chain visibility practices
- [x] Add GitHub Actions workflow
- [x] Test local generation successfully

## Summary by CodeRabbit

* **New Features**
  * Added automated SBOM generation and publication in CycloneDX and SPDX formats on release and tag updates, with SBOM files attached to releases and saved as build artifacts for extended retention.

* **Documentation**
  * Added SBOM security documentation with format details and example commands to inspect and analyze dependency components.

* **Tests**
  * Added a comprehensive API integration test suite covering health, authentication, ingest/analyzer flows, validation edge cases, event retrieval, exports, and basic performance expectations.

### CODE REVIEW

File: .github/workflows/sbom.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    tags:
+      - 'v*'
+  workflow_dispatch:
+
+jobs:
+  generate-sbom:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v4
```
[the remark this answers is not in the document]
Comment (Preetham404) [reply .github/workflows/sbom.yml]: The repository appears to use `actions/checkout@v6` in other workflows. Could you update this workflow to use the same version for consistency?

File: .github/workflows/sbom.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    
+    - name: Setup Python
+      uses: actions/setup-python@v5
```
[the remark this answers is not in the document]
Comment (Preetham404) [reply .github/workflows/sbom.yml]: The repository appears to use `actions/setup-python@v6` elsewhere. Could you update this workflow to align with the existing convention?

File: .github/workflows/sbom.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      with:
+        python-version: '3.11'
+    
+    - name: Generate SBOM
+      run: python scripts/generate_sbom.py
+    
+    - name: Upload to Release
+      uses: softprops/action-gh-release@v2
```
[the remark this answers is not in the document]
Comment (Preetham404) [reply .github/workflows/sbom.yml]: This workflow currently references `softprops/action-gh-release@v2`. Could you consider pinning the action to a specific commit SHA to improve supply chain security and align with GitHub Actions security best practices?

File: .github/workflows/sbom.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    
+    - name: Upload as Artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: sbom
+        path: |
+          sbom_cyclonedx.json
+          sbom_spdx.json
+        retention-days: 90
```
[the remark this answers is not in the document]
Comment (Preetham404) [reply .github/workflows/sbom.yml:46]: Thanks for the detailed review. Could you please address the security recommendations, particularly action pinning, scoped `GITHUB_TOKEN` permissions, and the `persist-credentials: false` setting where applicable?

File: docs/security/sbom.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+\# Software Bill of Materials (SBOM)
```
[the remark this answers is not in the document]
Comment (Preetham404) [reply docs/security/sbom.md]: Could you remove the leading backslash before the `#` so the heading renders correctly?

File: docs/security/sbom.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+| SPDX | `sbom\_spdx.json` | Linux Foundation | Compliance, Legal review |
+
+
+
+\## 📊 Current Dependencies
+
+\- \*\*Python packages:\*\* 18 (FastAPI, scikit-learn, pandas, etc.)
+
+\- \*\*Node.js packages:\*\* 17 (React, Vite, Recharts, etc.)
+
```
[the remark this answers is not in the document]
Comment (Preetham404) [reply docs/security/sbom.md]: The documentation appears to contain unnecessary escape characters. Could you remove the extra backslashes so the Markdown renders correctly and remains easy to read?

File: docs/security/sbom.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+\## 🔍 Viewing SBOM Contents
+
+
+
+```bash
+
+\# View all dependencies
+
+cat sbom\_cyclonedx.json | jq '.components\[] | {name, version}'
```
[the remark this answers is not in the document]
Comment (Preetham404) [reply docs/security/sbom.md]: The jq expression appears to contain escaped brackets that may result in invalid syntax. Could you verify the command and update it if necessary?

**Review by Preetham404 — COMMENTED** [review]
Thank you for the contribution. I have reviewed the changes and left a few comments regarding workflow consistency, security best practices, and documentation formatting. Please take a look and address them where applicable.

File: tests/integration/test_api_integration.py [hunk 8]
```diff
@@ -0,0 +1,829 @@
...
+
+        get_r = requests.get(
+            f"{INGEST_URL}/v1/events/{event_id}",
+            headers=VALID_HEADERS,
+            timeout=5,
+        )
+        # 200 if Redis is up, 503 if degraded — both acceptable
+        assert get_r.status_code in (200, 503, 404)
+        if get_r.status_code == 200:
+            assert get_r.json()["event_id"] == event_id
```
[the remark this answers is not in the document]
Comment (Preetham404) [reply tests/integration/test_api_integration.py:577]: The fixed `sleep` may introduce timing-related test failures. Could you consider replacing it with a retry-based approach?

File: tests/integration/test_api_integration.py [hunk 9]
```diff
@@ -0,0 +1,829 @@
...
+            if r.status_code != 200:
+                failures.append(f"Request {i} ({prompt!r}): {r.status_code}")
+
+        assert not failures, "Some batch requests failed:\n" + "\n".join(failures)
+
+    def test_health_check_response_under_500ms(self):
+        start = time.time()
+        requests.get(f"{INGEST_URL}/health", timeout=5)
+        elapsed = time.time() - start
+        assert elapsed < 3.0, f"Health check took {elapsed:.2f}s"
```
[the remark this answers is not in the document]
Comment (Preetham404) [reply tests/integration/test_api_integration.py:767]: The test names appear to imply stricter performance thresholds than the assertions currently enforce. Could you align the assertions with the documented expectations or update the test names to reflect the actual threshold being tested?

[the remark this answers is not in the document]
Comment (Preetham404) [reply tests/integration/test_api_integration.py:767]: The assertion threshold appears to be inconsistent with the test name. Could you align the assertion with the intended performance target or update the test name to reflect the actual threshold being validated?

File: tests/integration/test_api_integration.py [hunk 10]
```diff
@@ -0,0 +1,829 @@
...
+        assert r.status_code == 404
+
+    def test_404_on_unknown_analyzer_route(self):
+        r = requests.get(f"{ANALYZER_URL}/v1/nonexistent", headers=VALID_HEADERS, timeout=5)
+        assert r.status_code == 404
+
+    def test_405_on_wrong_method_ingest(self):
+        """GET on a POST-only endpoint should return 405."""
+        r = requests.get(f"{INGEST_URL}/v1/events/llm", headers=VALID_HEADERS, timeout=5)
+        assert r.status_code in (405, 503)
```
[the remark this answers is not in the document]
Comment (Preetham404) [reply tests/integration/test_api_integration.py:788]: Could you confirm whether this endpoint is expected to return `405` consistently? If so, the assertion may need to be tightened to validate the intended behavior.

**Review by Preetham404 — COMMENTED** [review]
Thank you for the contribution. I reviewed the changes and left a few comments regarding test reliability, assertion consistency, and validation behavior. Please review and address them where applicable.

**Comment by Preetham404** [comment]
I checked the failing TENET Security Review workflow. The failure appears to be due to a missing `TENET_AI_KEY` repository secret rather than an issue introduced by this PR. This may require maintainer action to configure the workflow secret.

**Comment by S3DFX-CYBER** [comment]
@Aharshi3614 needs fixes as requested by mentor

**Comment by S3DFX-CYBER** [comment]
@Aharshi3614 needs fixes as per the bot

**Comment by S3DFX-CYBER** [comment]
@coderabbitai is pr ready to be merged?

**Comment by S3DFX-CYBER** [comment]
Closing due to inactivity
